src/ast/mod.rs

```diff
@@ -146,7 +146,38 @@ impl FunctionCtx {
             }
             Call(ref func, ref args) => unimplemented!(),
             Index(ref tup, ref idx) => unimplemented!(),
-            Mktup(ref parts) => unimplemented!(),
+            Mktup(ref parts) => {
+                let reg = self.push_tmp();
+                let mut code = vec![];
+                let mut part_addrs = vec![];
+                let mut start_addr = None;
+                // @TODO: any way to make start_addr not mutable?
+                for (i, part) in parts.iter().enumerate() {
+                    let (part_dest, mut part_code) = self.compile_expr(part);
+                    code.append(&mut part_code);
+                    part_addrs.push(part_dest);
+
+                    if i == 0 {
+                        start_addr = Some(part_dest);
+                    }
+                }
+
+                assert!(start_addr.is_some());
```

Comment:
This is unable to handle empty tuples (although I guess our bytecode instruction is also unable to handle empty tuples?)

Reply:
In any case, we should handle there being no `start_addr` and allocate one if necessary.